### PR TITLE
Add test for zero sized index buffer out of bounds draw

### PR DESCRIPTION
**Pull request by shrekshao**
Add a new section for testing zero sized index buffer out of bounds draw

-----

- [x] New helpers, if any, are documented in `helper_index.md`.
- [x] Incomplete tests, if any, are marked with TODO or `.unimplemented()`.

**[Review requirement](https://github.com/gpuweb/cts/blob/main/docs/reviews.md) checklist:**

- [ ] WebGPU readability
- [ ] TypeScript readability

### CODE REVIEW

**Review by austinEng — APPROVED** [review]
LGTM

File: src/webgpu/api/validation/encoding/cmds/index_access.spec.ts [hunk 1]
```diff
@@ -149,9 +146,44 @@ g.test('out_of_bounds')
...
+
+g.test('out_of_bounds_zero_sized_index_buffer')
+  .cases(pbool('indirect'))
+  .subcases(
+    () =>
+      params()
+        .combine([
+          { indexCount: 3, firstIndex: 1 }, // indexCount + firstIndex out of bound
+          { indexCount: 0, firstIndex: 1 }, // indexCount is 0 but firstIndex out of bound
+          { indexCount: 3, firstIndex: 0 }, // only firstIndex out of bound
```
Comment (kainino0x) [inline src/webgpu/api/validation/encoding/cmds/index_access.spec.ts:170]:
```suggestion
          { indexCount: 3, firstIndex: 0 }, // only indexCount out of bound
```
?

File: src/webgpu/api/validation/encoding/cmds/index_access.spec.ts [hunk 2]
```diff
@@ -149,9 +146,44 @@ g.test('out_of_bounds')
...
+        indexBuffer,
+        new Uint32Array([indexCount, instanceCount, firstIndex, 0, 0]),
+        0
+      );
+    } else {
+      t.drawIndexed(indexBuffer, indexCount, instanceCount, firstIndex, 0, 0);
+    }
+  });
+
+g.test('out_of_bounds_zero_sized_index_buffer')
```
Comment (kainino0x) [inline src/webgpu/api/validation/encoding/cmds/index_access.spec.ts:162]: Please add a description (info: https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md)

**Comment by shrekshao** [comment]
Thanks for the review. Fixed the comments
